### PR TITLE
Added Theme Table container and base for Tailwind Custom Themes

### PR DESCRIPTION
**Pull request by marcheffels**
## ⚡ PowerGrid - Pull Request

Welcome and thank you for your interest in contributing to our project!. You must use this template to submit a Pull Request or it will not be accepted.

--- 
#### Motivation

- [ ] Bug fix
- [ ] Enhancement
- [x] New feature
- [ ] Breaking change

#### Description

This Pull Request adds Tailwind Custom Theme container and base.

#### Documentation

 This PR requires [Documentation](https://github.com/Power-Components/powergrid-doc) update?

- [ ] Yes
- [x] No
- [ ] I have already submitted a Documentation pull request.

### CODE REVIEW

**Comment by luanfreitasdev** [comment]
Thank you @marcheffels

**Comment by luanfreitasdev** [comment]
I'm afraid this is a breaking change. If the theme was previously published and the class was not extended, there will be a break.

**Comment by marcheffels** [comment]
Hi @luanfreitasdev, I'm understanding why you cannot marge this PR.

The problem I'm trying to solve is that the current theme is breaking my site.
The negative margin on the container causes the container size to be larger than the parent div.

My current workaround for my site is that I wrap all the tables in a separate div with `overflow-hidden`.

### Suggested solutions:
**1. Remove negative margin:**
Container changes:
From `class="-my-2 overflow-x-auto sm:-mx-3 lg:-mx-8"`
To `class="overflow-x-auto"`
Base changes:
From `class="p-3 align-middle inline-block min-w-full w-full sm:px-6 lg:px-8"`
To `class="align-middle inline-block min-w-full w-full"`
(I tried to find a reason for the negative margin. If there is a reason, please explain it to me.)

**2. Add "overflow-hidden" to the wrapper:**
Change parent of container:
From `class="flex flex-col"`
To `class="flex flex-col overflow-hidden"`

Are any of the suggested solutions acceptable?
Then I can create a new PR.

Thanks.

File: resources/views/components/frameworks/tailwind/table-base.blade.php [hunk 1]
```diff
@@ -4,11 +4,13 @@ class="flex flex-col"
 >
     <div
         id="power-grid-table-container"
-        class="-my-2 overflow-x-auto sm:-mx-3 lg:-mx-8"
+        class="{{ data_get($theme, 'table.containerClass') }}"
```
Comment (luanfreitasdev) [inline resources/views/components/frameworks/tailwind/table-base.blade.php]: Add default values to all data_get methods

```php
class="{{ data_get($theme, 'table.containerClass', '-my-2 overflow-x-auto sm:-mx-3 lg:-mx-8') }}"
```

**Review by luanfreitasdev — CHANGES_REQUESTED** [review]
Can you check for a bootstrap theme too?

File: resources/views/components/frameworks/tailwind/table-base.blade.php (hunk 1 above)
Comment (marcheffels) [reply resources/views/components/frameworks/tailwind/table-base.blade.php]: The `data_get` function returns the default value only if the specified key is not found. The `containerClass` and all others are defined as public properties with an initially empty string. Therefore, `data_get` never returns the default value here.

Comment (luanfreitasdev) [reply resources/views/components/frameworks/tailwind/table-base.blade.php]: This is necessary to maintain compatibility with other themes that do not use the new methods. For example: 

https://github.com/Power-Components/powergrid-demo/blob/powergrid_v5/app/Helpers/PowerGridThemes/TailwindStriped.php

**Comment by marcheffels** [comment]
> Can you check for a bootstrap theme too?

in Bootstrap theme I can't find any container, base div tag or negative margin, so I believe this problem exists only in Tailwind theme.

**Review by luanfreitasdev — APPROVED** [review]

**Pull request merged**